remote-view/shared/src/main/scala/parsley/debug/RemoteView.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2023 Parsley Debug View Contributors <https://github.com/j-mie6/parsley-debug-views/graphs/contributors>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+package parsley.debug
+
+import parsley.debug.internal.DebugTreeSerialiser
+
+import sttp.client3.*
+
+// import scala.util.Try
+import scala.concurrent.duration.*
+import sttp.model.Uri
+import scala.util.Try
+import scala.util.Failure
+import scala.util.Success
+
+/** The RemoteView HTTP module allows the parsley debug tree to be passed off to a server through a specified port on
+  * local host (by default) or to a specified IP address. This enables all of the debug tree parsing, serving and 
+  * graphics to be done in a separate process, and served reactively to a client.
+  * 
+  * This module is part of the Debugging Interactively in parsLey Library (DILL) project,
+  * (https://github.com/j-mie6/parsley-debug-app).
+  * 
+  * RemoteView uses the STTP library to create HTTP requests to a specified IP address, over a specified port.
+  * The request is formatted using the upickle JSON formatting library, it is being used over other
+  * libraries like circe for its improved speed over large data structures.
+  */
+sealed trait RemoteView extends DebugView.Reusable {
+  protected val port: Int
+  protected val address: String
+
+  // Printing helpers
+  private [debug] final val TextToRed: String    = "\u001b[31m"
+  private [debug] final val TextToNormal: String = "\u001b[0m"
+  
+  // Request Timeouts
+  private [debug] final val ConnectionTimeout: FiniteDuration = 30.second
+  private [debug] final val ResponseTimeout: FiniteDuration = 10.second
+
+  /**
+   * Send the debug tree and input to the port and address specified in the 
+   * object construction. 
+   *
+   * @param input The input source.
+   * @param tree The debug tree.
+   */
+  override private [debug] def render(input: => String, tree: => DebugTree): Unit = {
+    // Endpoint for post request
+    val endPoint: Uri = uri"http://$address:$port/api/remote"
+    // JSON formatted payload for post request
+    val payload: String = DebugTreeSerialiser.toJSON(input, tree)
+    
+    // Send POST
+    println("Sending Debug Tree to Server")
+    
+    val backend = TryHttpURLConnectionBackend(
+      options = SttpBackendOptions.connectionTimeout(ConnectionTimeout)
+    )
+
+    val response: Try[Response[Either[String,String]]] = basicRequest
+      .readTimeout(ResponseTimeout)
+      .header("User-Agent", "remoteView")
+      .contentType("application/json")
+      .body(payload)
+      .post(endPoint)
+      .send(backend)
+
+    response match {
+      case Failure(exception) => println(s"${TextToRed}Remote View request failed! Please validate address ($address) and port number ($port).${TextToNormal}\n\tError : ${exception.toString()}")
+      case Success(value) => value.body match {
```

Comment:
```suggestion
      case Success(res) => res.body match {
```
or `response`, I guess. Bit clearer what it is that way

Reply:
👍